carbon-design-system/carbon-platform
# feat(web-app): Resolve design kit schema references

Closes #970 

Resolves and validates designKits at library level, introduces all design kits function + filter through allowlist

#### Changelog

**New**

- getAllDesignKits global function

**Changed**

- getLibraryData: validate design kits, resolve references



#### Testing / reviewing

http://localhost:3000/design-kits: 
- should have all DKs registered in https://github.com/carbon-design-system/carbon-platform/blob/main/packages/resources/carbon.yml
- should have `added-design-kit-2` from https://github.com/francinelucca/library-testing/blob/main/carbon.yml 
- added-design-kit-1 (from https://github.com/francinelucca/library-testing/blob/main/carbon.yml) should be skipped since it's not on allow list, see terminal logs

http://localhost:3000/libraries/design-kit:
- see designKits list by scrolling all the way down, should have all designKits defined in library.designKits in https://github.com/francinelucca/library-testing/blob/main/carbon.yml

http://localhost:3000/libraries/invalid-kits-library:
- see designKits list by scrolling all the way down, should have first 5 designKits defined in library.designKits in https://github.com/francinelucca/library-testing/blob/main/folder/carbon.yml
- the last 4 designKits defined in library.designKits in https://github.com/francinelucca/library-testing/blob/main/folder/carbon.yml should be skipped since they're invalid, see terminal logs

Note: I will remove all these functions and test data once approved

TODO:

- [x] Remove pages/design-kits.js
- [x] Remove test entries from libraryAllowList
- [x] Remove DesignKit source from data/design-kits.js
- [x] Remove Text design kit allowlist entry in data/design-kits.js
- [x] Remove libraryData designkits list in index page
